frontend_tests/node_tests/stream_topic_history.js

```diff
@@ -39,23 +39,51 @@ run_test("basics", () => {
     assert.deepEqual(history, ["topic2", "Topic1"]);
     assert.deepEqual(max_message_id, 103);
 
+    stream_topic_history.add_message({
+        stream_id,
+        message_id: 104,
+        topic_name: "Topic1",
+    });
+    history = stream_topic_history.get_recent_topic_names(stream_id);
+    max_message_id = stream_topic_history.get_max_message_id(stream_id);
+    assert.deepEqual(history, ["Topic1", "topic2"]);
+    assert.deepEqual(max_message_id, 104);
+
+    set_global("message_util", {
+        get_messages_in_topic: () => [{id: 101}, {id: 102}],
+    });
+    // Removing the last msg in topic1 changes the order
+    stream_topic_history.remove_messages({
+        stream_id,
+        topic_name: "Topic1",
+        num_messages: 1,
+        max_removed_msg_id: 104,
+    });
+    history = stream_topic_history.get_recent_topic_names(stream_id);
+    assert.deepEqual(history, ["topic2", "Topic1"]);
+    // Removing a topic message shouldn't effect the max_message_id.
+    max_message_id = stream_topic_history.get_max_message_id(stream_id);
+    assert.deepEqual(max_message_id, 104);
```

Comment:
I'm not sure why this is 104 -- is this because it's the stream-level value, not the topic-level value?  I'd like us to assert that the top-level values changed as expected somewhere in this test block, since that's one of the important pieces of logic to verify.

Reply:
Yeah, it is stream-level value.